vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -102,7 +102,9 @@ def __init__(
 
     def prepare_cos_sin(self,
                         positions: torch.Tensor,
-                        offsets: Optional[torch.Tensor] = None):
+                        offsets: Optional[torch.Tensor] = None,
+                        recompute_cos_sin: bool = False):
+        self.recompute_cos_sin = recompute_cos_sin
```

Comment:
I think if you set
self.recompute_cos_sin = os.getenv('VLLM_COS_SIN_RECOMPUTE', 'false').lower() in ['1', 'true'])
in the __init__ method, you don't have to pass the recompute_cos_sin parameter here (see my other comment on hpu_model_runner.py:753)

Reply:
I think the 'RotaryEmbedding.__init__()' is general for all HW, so I am thinking to only pass this new 'argument' in prepare_cos_sin() which is added by us?